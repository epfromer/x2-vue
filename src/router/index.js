import Vue from 'vue'
import VueRouter from 'vue-router'
import AppSettingsView from '../views/AppSettingsView.vue'
import BarView from '../views/BarView.vue'
import ChordView from '../views/ChordView.vue'
import EmailDetailView from '../views/EmailDetailView.vue'
import EventTimelineView from '../views/EventTimelineView.vue'
import HomeView from '../views/HomeView.vue'
import NetworkGraphView from '../views/NetworkGraphView.vue'
import PieView from '../views/PieView.vue'
import PolarView from '../views/PolarView.vue'
import SavedSearchView from '../views/SavedSearchView.vue'
import SearchView from '../views/SearchView.vue'
import TreeMapView from '../views/TreeMapView.vue'
import VolumeTimelineView from '../views/VolumeTimelineView.vue'
import WordCloudView from '../views/WordCloudView.vue'

Vue.use(VueRouter)

export const routes = [
  {
    path: '/AppSettingsView',
    name: 'AppSettingsView',
    component: AppSettingsView,
  },
  { path: '/SearchView', name: 'SearchView', component: SearchView },
  {
    path: '/EmailDetailView/:id',
    name: 'EmailDetailView',
    component: EmailDetailView,
    props: true,
  },
  {
    path: '/SavedSearchView',
    name: 'SavedSearchView',
    component: SavedSearchView,
  },
  { path: '/ChordView', name: 'ChordView', component: ChordView },
  { path: '/PieView', name: 'PieView', component: PieView },
  {
    path: '/NetworkGraphView',
    name: 'NetworkGraphView',
    component: NetworkGraphView,
  },
  { path: '/WordCloudView', name: 'WordCloudView', component: WordCloudView },
  {
    path: '/VolumeTimelineView',
    name: 'VolumeTimelineView',
    component: VolumeTimelineView,
  },
  { path: '/TreeMapView', name: 'TreeMapView', component: TreeMapView },
  {
    path: '/EventTimelineView',
    name: 'EventTimelineView',
    component: EventTimelineView,
  },
  {
    path: '/BarView',
    name: 'BarView',
    component: BarView,
  },
  {
    path: '/PolarView',
    name: 'PolarView',
    component: PolarView,
  },
  { path: '/', name: 'HomeView', component: HomeView },
]

export const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
})

export default router
