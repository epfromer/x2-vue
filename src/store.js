import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    // current page of emails
    listEmails: [],

    // total emails that match query
    totalEmails: 0
  },
  mutations: {
    setEmails(state, emails) {
      state.listEmails = emails.listEmails;
      state.totalEmails = emails.totalEmails;
    }
  },
  actions: {
    setEmailsAsync: (context, emails) => context.commit('setEmails', emails)
  }
});
