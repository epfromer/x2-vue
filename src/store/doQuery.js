function encodeQuery(q) {
  // encode query for URL
  let params = ''
  Object.keys(q).forEach((key) => {
    if ((typeof q[key] == 'string' && q[key]) || typeof q[key] == 'number') {
      params += '&' + key + '=' + q[key]
    }
  })
  return '?' + params.slice(1)
}

export default async function doQuery(
  setVuexState,
  emailListPage,
  emailListItemsPerPage,
  querySort,
  queryOrder,
  sent,
  timeSpan,
  from,
  to,
  subject,
  allText,
  queryStart,
  queryFinish
) {
  // construct query
  const query = {
    skip: emailListPage * emailListItemsPerPage,
    limit: emailListItemsPerPage,
    sort: querySort,
    order: queryOrder,
  }
  if (sent) query.sent = sent
  if (timeSpan) query.timeSpan = timeSpan
  if (from) query.from = from
  if (to) query.to = to
  if (subject) query.subject = subject
  if (allText) query.allText = allText

  const encodedQuery = encodeQuery(query)

  // do fetch
  queryStart()
  const url = `${process.env.VUE_APP_EMAIL_SERVER}/email/${encodedQuery}`
  console.log(url)
  const resp = await fetch(url)
  resp
    .json()
    .then((resp) => {
      console.log('setting vuex state')
      setVuexState('emails', resp.listDocs)
      setVuexState('totalEmails', resp.total)
    })
    .catch(() => {}) // TODO: handle errors
    .then(queryFinish())
  // .then(() => console.log('loading complete'))
}
