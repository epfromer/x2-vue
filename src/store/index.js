import request, { gql } from 'graphql-request'
import _ from 'lodash'
import Vue from 'vue'
import Vuex from 'vuex'
import { defaultLimit } from './constants'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    // appSettingsSlice
    darkMode: localStorage.getItem('darkMode') === 'true' ? true : false,
    themeName: '',
    // TODO remove
    densePadding:
      localStorage.getItem('densePadding') === 'false' ? false : true,
    themePrimaryColor: localStorage.getItem('themePrimaryColor')
      ? localStorage.getItem('themePrimaryColor')
      : '#2196f3',
    themeSecondaryColor: localStorage.getItem('themeSecondaryColor')
      ? localStorage.getItem('themeSecondaryColor')
      : '#f50057',

    // authenticationSlice
    authenticated: false,
    username: '',

    // custodiansSlice
    custodiansLoading: false,
    custodians: undefined,

    // emailSentByDaySlice
    emailSentByDayLoading: false,
    emailSentByDay: undefined,

    // emailSlice
    emailLoading: false,
    email: undefined,
    emailTotal: 0,

    // querySlice
    sort: 'sent',
    order: 1,
    sent: '',
    from: '',
    to: '',
    subject: '',
    allText: '',
    body: '',
    emailListPage: 0,

    // wordCloudSlice
    wordCloudLoading: false,
    wordCloud: undefined,
  },
  getters: {
    // component suse mapState to get to simple state items
    // these getters are for more complex state manipulation

    // appSettingsSlice

    // authenticationSlice

    // custodiansSlice
    getEmailSenders: (state) => {
      const custodians = state.custodians.custodians
      const data = []
      if (state.custodians) {
        state.custodians.forEach((custodian) => {
          if (custodian.senderTotal) {
            data.push({
              name: custodian.name,
              value: custodian.senderTotal,
              color: custodian.color,
            })
          }
        })
      }
      return data
    },
    getEmailReceivers: (state) => {
      const custodians = state.custodians.custodians
      const data = []
      if (state.custodians) {
        state.custodians.forEach((custodian) => {
          if (custodian.receiverTotal) {
            data.push({
              name: custodian.name,
              value: custodian.receiverTotal,
              color: custodian.color,
            })
          }
        })
      }
      return data
    },
    getEmailSentByCustodian: (state) => {
      const custodianNameFromId = (id) => {
        if (state.custodians && state.custodians) {
          const c = state.custodians.find((c) => c.id === id)
          return c ? c.name : ''
        }
        return ''
      }

      const custodians = state.custodians
      const data = []
      const nodes = []

      if (custodians) {
        custodians.forEach((fromCustodian) => {
          fromCustodian.toCustodians.forEach((toCustodian) => {
            data.push({
              source: fromCustodian.name,
              target: custodianNameFromId(toCustodian.custodianId),
              value: toCustodian.total,
            })
          })
        })
        // and array of color keys
        custodians.forEach((custodian) => {
          nodes.push({ id: custodian.name, color: custodian.color })
        })
      }

      return { data, nodes }
    },
    getContactColor: (state) => (name) =>
      state.contacts.find((c) => c.name === name).color,

    // emailSentByDaySlice

    // emailSlice

    // querySlice
    getTo: (state) => state.emailLoading,
    getQuery: (state) => {
      const query = {
        skip: state.emailListPage * defaultLimit,
        limit: defaultLimit,
        sort: state.sort,
        order: state.order,
      }
      if (state.sent) query.sent = state.sent
      if (state.from) query.from = state.from
      if (state.to) query.to = state.to
      if (state.subject) query.subject = state.subject
      if (state.allText) query.allText = state.allText
      if (state.body) query.body = state.body
      return query
    },
    getEmailById: (state) => (id) => state.emails.find((e) => e._id === id),
    getNextEmail: (state) => (id) => {
      if (!state.emails || !state.emails.length) return undefined
      const i = state.emails.findIndex((e) => e._id === id)
      return i < state.emails.length - 1 ? state.emails[i + 1] : undefined
    },
    getPreviousEmail: (state) => (id) => {
      if (!state.emails || !state.emails.length) return undefined
      const i = state.emails.findIndex((e) => e._id === id)
      return i > 0 ? state.emails[i - 1] : undefined
    },
    getEmailIndex: (state) => (id) =>
      state.emails.findIndex((e) => e._id === id) + 1,

    // wordCloudSlice
    getWordCloudLoading: (state) => state.wordCloudLoading,
    getWordCloud: (state) => state.wordCloud,
  },
  mutations: {
    // appSettingsSlice
    setDarkMode: (state, darkMode) => {
      state.darkMode = darkMode
    },
    setThemeName: (state, themeName) => {
      state.themeName = themeName
    },

    // authenticationSlice
    setAuthenticated: (state, authenticated) => {
      state.authenticated = authenticated
    },
    setUsername: (state, username) => {
      state.username = username
    },

    // custodiansSlice
    setCustodiansLoading: (state, custodiansLoading) => {
      state.custodiansLoading = custodiansLoading
    },
    setCustodians: (state, custodians) => {
      state.custodians = _.cloneDeep(custodians)
    },

    // emailSentByDaySlice
    setEmailSentByDayLoading: (state, emailSentByDayLoading) => {
      state.emailSentByDayLoading = emailSentByDayLoading
    },
    setEmailSentByDay: (state, emailSentByDay) => {
      state.emailSentByDay = _.cloneDeep(emailSentByDay)
    },

    // emailSlice
    setEmailLoading: (state, emailLoading) => {
      state.emailLoading = emailLoading
    },
    setEmail: (state, email) => {
      state.email = _.cloneDeep(email)
    },
    appendEmail: (state, email) => {
      if (state.email) {
        // console.log('appending email')
        state.email.push(...email)
      } else {
        // console.log('setting email')
        state.email = _.cloneDeep(email)
      }
    },
    setEmailTotal: (state, emailTotal) => {
      state.emailTotal = emailTotal
    },

    // querySlice
    setSort: (state, sort) => {
      state.sort = sort
    },
    setOrder: (state, order) => {
      state.order = order
    },
    setSent: (state, sent) => {
      state.sent = sent
    },
    setFrom: (state, from) => {
      state.from = from
    },
    setTo: (state, to) => {
      state.to = to
    },
    setSubject: (state, subject) => {
      state.subject = subject
    },
    setAllText: (state, allText) => {
      state.allText = allText
    },
    setBody: (state, body) => {
      state.body = body
    },
    setEmailListPage: (state, emailListPage) => {
      state.emailListPage = emailListPage
    },
    clearSearch: (state) => {
      state.sent = ''
      state.timeSpan = 0
      state.from = ''
      state.to = ''
      state.subject = ''
      state.allText = ''
      state.body = ''
    },

    // wordCloudSlice
    setWordCloud: (state, wordCloud) => {
      state.wordCloud = _.cloneDeep(wordCloud)
    },
    setWordCloudLoading: (state, wordCloudLoading) => {
      state.wordCloudLoading = wordCloudLoading
    },
  },
  actions: {
    // https://vuex.vuejs.org/guide/actions.html - async mutations

    // appSettingsSlice
    loadAppSettingsAsync: async ({ commit, store }) => {
      try {
        let darkMode = false
        let themeName = defaultThemeName
        if (typeof Storage !== 'undefined') {
          if (
            localStorage.getItem('darkMode') &&
            localStorage.getItem('darkMode') === 'true'
          ) {
            darkMode = true
          }
          if (
            localStorage.getItem('themeName') &&
            localStorage.getItem('themeName') !== 'null'
          ) {
            themeName = localStorage.getItem('themeName')
          }
        } else {
          const value = await AsyncStorage.getItem('darkMode')
          if (value === 'true') darkMode = true
          value = await AsyncStorage.getItem('themeName')
          if (value) themeName = value
        }
        commit('setDarkMode', darkMode)
        commit('setThemeName', themeName)
      } catch (e) {
        console.error(e)
      }
    },
    setDarkModeAsync: async ({ commit }, darkMode) => {
      if (typeof Storage !== 'undefined') {
        localStorage.setItem('darkMode', String(darkMode))
      } else {
        await AsyncStorage.setItem('darkMode', String(darkMode))
      }
      commit('setDarkMode', darkMode)
    },
    setThemeNameAsync: async ({ commit }, themeName) => {
      if (typeof Storage !== 'undefined') {
        localStorage.setItem('themeName', themeName)
      } else {
        await AsyncStorage.setItem('themeName', themeName)
      }
      commit('setThemeName', themeName)
    },

    // authenticationSlice

    // custodiansSlice
    getCustodiansAsync: ({ commit }) => {
      commit('setCustodiansLoading', true)
      const server = process.env.VUE_APP_X2_SERVER
      const query = gql`
        {
          getCustodians {
            id
            name
            title
            color
            senderTotal
            receiverTotal
            toCustodians {
              custodianId
              total
            }
          }
        }
      `
      return request(`${server}/graphql/`, query)
        .then((data) => {
          commit('setCustodians', data.getCustodians)
          commit('setCustodiansLoading', false)
        })
        .catch((e) => console.error(e))
    },

    // emailSentByDaySlice

    // emailSlice
    getEmailAsync: ({ commit, getters }, append = false) => {
      commit('setEmailLoading', true)
      const server = process.env.VUE_APP_X2_SERVER
      // console.log(server)
      const query = gql`
        query getEmail(
          $skip: Int
          $limit: Int
          $sort: String
          $order: Int
          $sent: String
          $from: String
          $to: String
          $subject: String
          $allText: String
          $body: String
        ) {
          getEmail(
            skip: $skip
            limit: $limit
            sort: $sort
            order: $order
            sent: $sent
            from: $from
            to: $to
            subject: $subject
            allText: $allText
            body: $body
          ) {
            emails {
              id
              sent
              sentShort
              from
              fromCustodian
              to
              toCustodians
              cc
              bcc
              subject
              body
            }
            total
          }
        }
      `
      return request(`${server}/graphql/`, query, getters.getQuery)
        .then((data) => {
          if (append) {
            commit('appendEmail', data.getEmail.emails)
          } else {
            commit('setEmail', data.getEmail.emails)
          }
          commit('setEmailTotal', data.getEmail.total)
          commit('setEmailLoading', false)
        })
        .catch((e) => console.error(e))
    },

    // querySlice

    // wordCloudSlice
    getInitialDataAsync: ({ commit }) => {
      commit('setCustodiansLoading', true)
      commit('setEmailSentByDayLoading', true)
      commit('setWordCloudLoading', true)
      const server = process.env.VUE_APP_X2_SERVER
      const query = gql`
        {
          getWordCloud {
            tag
            weight
          }
          getEmailSentByDay {
            sent
            total
          }
          getCustodians {
            id
            name
            title
            color
            senderTotal
            receiverTotal
            toCustodians {
              custodianId
              total
            }
          }
        }
      `
      return request(`${server}/graphql/`, query)
        .then(async (data) => {
          // await sleep(5000)
          commit('setCustodians', data.getCustodians)
          commit('setCustodiansLoading', false)
          commit('setEmailSentByDay', data.getEmailSentByDay)
          commit('setEmailSentByDayLoading', false)
          commit('setWordCloud', data.getWordCloud)
          commit('setWordCloudLoading', false)
        })
        .catch((e) => console.error(e))
    },
  },
})
