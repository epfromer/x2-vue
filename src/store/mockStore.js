export const store = {
  getters: {
    getEmailById: () => () => ({
      _id: '5ccedc7214a5b6cd9203206f',
      body: 'Thanks for your interest in the Assistant Trader',
      clientSubmitTime: '2001-10-29T22:55:22.000Z',
      displayBCC: '',
      displayCC: 'Sager  Elizabeth',
      displayTo: 'Samuel Behrends',
      senderEmailAddress: 'Marcus.Nettelton@ENRON.com',
      senderName: 'Nettelton  Marcus',
      subject: 'RE: Draft of Market rate filing',
    }),
    getNextEmail: () => () => ({
      _id: '5ec911ddfef4030598a9347b',
      body: 'I feel that I have become a valuable, knowledgeable member',
      clientSubmitTime: '2001-10-29T22:55:22.000Z',
      displayBCC: '',
      displayCC: 'Sager  Elizabeth',
      displayTo: 'Samuel Behrends',
      senderEmailAddress: 'Marcus.Nettelton@ENRON.com',
      senderName: 'Nettelton  Marcus',
      subject: 'RE: Draft of Market rate filing',
    }),
    getPreviousEmail: () => () => ({
      _id: '5ccedc7214a5b6cd9203206d',
      body: 'We are continuing work to ensure our new online trading system',
      clientSubmitTime: '2001-10-29T23:17:34.000Z',
      displayBCC: '',
      displayCC: 'Catherine McCarthy; Sager  Elizabeth',
      displayTo: 'Samuel Behrends',
      senderEmailAddress: 'Marcus.Nettelton@ENRON.com',
      senderName: 'Nettelton  Marcus',
      subject: 'RE: Draft of Market rate filing',
    }),
    getEmailIndex: () => () => 1,
    getContactColor: () => () => '#121212',
  },
  mutations: {
    setVuexState: jest.fn(),
    saveAppSettings: jest.fn(),
    clearSearch: jest.fn(),
  },
  actions: {},
  state: {
    // search results
    emails: [
      {
        _id: '5ccedc7214a5b6cd9203206d',
        body: 'We are continuing work to ensure our new online trading system',
        clientSubmitTime: '2001-10-29T23:17:34.000Z',
        displayBCC: '',
        displayCC: 'Catherine McCarthy; Sager  Elizabeth',
        displayTo: 'Samuel Behrends',
        senderEmailAddress: 'Marcus.Nettelton@ENRON.com',
        senderName: 'Nettelton  Marcus',
        subject: 'RE: Draft of Market rate filing',
      },
      {
        _id: '5ccedc7214a5b6cd9203206f',
        body: 'Thanks for your interest in the Assistant Trader',
        clientSubmitTime: '2001-10-29T22:55:22.000Z',
        displayBCC: '',
        displayCC: 'Sager  Elizabeth',
        displayTo: 'Samuel Behrends',
        senderEmailAddress: 'Marcus.Nettelton@ENRON.com',
        senderName: 'Nettelton  Marcus',
        subject: 'RE: Draft of Market rate filing',
      },
      {
        _id: '5ec911ddfef4030598a9347b',
        body: 'I feel that I have become a valuable, knowledgeable member',
        clientSubmitTime: '2001-10-29T22:55:22.000Z',
        displayBCC: '',
        displayCC: 'Sager  Elizabeth',
        displayTo: 'Samuel Behrends',
        senderEmailAddress: 'Marcus.Nettelton@ENRON.com',
        senderName: 'Nettelton  Marcus',
        subject: 'RE: Draft of Market rate filing',
      },
    ],
    totalEmails: 3,

    // email list
    emailListPage: 1,
    emailListItemsPerPage: 5,

    // query
    querySort: 'sent',
    queryOrder: 1,
    sent: '',
    timeSpan: 0,
    from: '',
    to: '',
    subject: '',
    allText: 'rate',
    body: '',

    // stats
    emailSentLoading: false,
    emailSent: [
      {
        sent: '1999-01-06',
        ids: ['301385bc-8730-4c5f-9d44-094fb4791352'],
      },
      {
        sent: '1999-01-07',
        ids: ['301385bc-8730-4c5f-9d44-094fb4791352'],
      },
    ],
    wordCloudLoading: false,
    wordCloud: [
      { tag: 'avici', weight: 28 },
      { tag: 'azurix', weight: 475 },
      { tag: 'backbone', weight: 161 },
      { tag: 'braveheart', weight: 27 },
    ],
    contactsLoading: false,
    contacts: [
      {
        _id: '5ec916d6fef4030598a9c835',
        senderTotal: 6,
        receiverTotal: 8,
        asSender: [
          {
            id: 'f8a62092-9ac1-4cfb-b7d4-dc828906da0b',
            to: ['Cash, Michelle'],
            sent: '2001-11-20T00:07:49.000Z',
          },
          {
            id: 'ea670da4-7e30-4296-abf4-293db35f9ff7',
            to: ['Buy, Richard'],
            sent: '2001-08-14T16:53:03.000Z',
          },
          {
            id: 'fd04ce13-2081-4c7b-9f8d-00c559925536',
            to: ['Buy, Richard'],
            sent: '2001-08-15T00:27:42.000Z',
          },
          {
            id: '3d7622e5-8598-440a-9320-456bbb0c325b',
            to: ['Buy, Richard'],
            sent: '2001-09-27T16:29:35.000Z',
          },
        ],
        asReceiver: [
          {
            id: 'd5b2ef5a-a027-489c-ad06-6def049b9f2d',
            from: 'Buy, Richard',
            sent: '2001-09-28T01:49:10.000Z',
          },
          {
            id: 'd5b2ef5a-a027-489c-ad06-6def049b9f2d',
            from: 'Buy, Richard',
            sent: '2001-09-28T01:49:10.000Z',
          },
        ],
        name: 'Watkins, Sherron',
        title: 'Vice President of Corporate Development',
        color: '#F06292',
        aliases: [
          'Sherron Watkins',
          'sherron',
          'Watkins Sherron',
          'watkins  sherron',
          'sherron_watkins@enron.net',
          'watkins',
        ],
      },
      {
        _id: '5ec916d6fef4030598a9c836',
        senderTotal: 0,
        receiverTotal: 164,
        asSender: [],
        asReceiver: [
          {
            id: 'f5a0e1bd-bf21-45af-9ead-9d0bff8c2501',
            from: 'Fleming, Rosalee',
            sent: '2000-12-13T19:24:00.000Z',
          },
          {
            id: '1d6b2d01-a893-4c64-bd6f-0246d450cc6a',
            from: 'Fleming, Rosalee',
            sent: '2000-12-14T02:04:00.000Z',
          },
          {
            id: '36b4d986-196f-4b32-868e-23dc864e86a2',
            from: 'Fleming, Rosalee',
            sent: '2000-12-14T02:02:00.000Z',
          },
          {
            id: '47c91a19-6c3b-4fca-a425-5485b07c4e6c',
            from: 'Fleming, Rosalee',
            sent: '2000-12-12T22:47:00.000Z',
          },
          {
            id: 'ce0b14ce-4e3f-4a00-a991-8fb99723e08e',
            from: 'Skilling, Jeff',
            sent: '2001-04-20T10:18:00.000Z',
          },
          {
            id: '28b4f565-2456-4607-8267-89d9a6e06b14',
            from: 'Lavorato, John',
            sent: '2000-10-10T23:35:00.000Z',
          },
          {
            id: '11ba6534-7cd6-46a7-9236-e5bc39a6d96e',
            from: 'Lavorato, John',
            sent: '1999-12-21T19:14:00.000Z',
          },
          {
            id: '3c5ed9d0-132b-4a66-bf7c-63b4a60b47cd',
            from: 'Fleming, Rosalee',
            sent: '2000-12-01T22:56:00.000Z',
          },
          {
            id: '08cac43d-70e2-415c-9f21-3ef61da479de',
            from: 'Fleming, Rosalee',
            sent: '1999-08-16T16:17:00.000Z',
          },
          {
            id: 'ec0fa1a7-9c29-4e41-bc75-0aabe9bfb794',
            from: 'Kitchen, Louise',
            sent: '2000-03-24T02:42:00.000Z',
          },
        ],
        name: 'Baxter, Cliff',
        title: 'Vice Chairman',
        color: '#5FC81D',
        aliases: [
          'cliff baxter',
          'cbaxter@enron.com',
          'Baxter@ect, Cliff',
          'Cliff, Baxter',
          'cliff baxter@ect',
          'baxter  cliff  aep',
          'baxter  cliff',
          'baxter',
          'cliff baxter/hou/ect@enron',
        ],
      },
      {
        _id: '5ec916d6fef4030598a9c838',
        senderTotal: 17,
        receiverTotal: 271,
        asSender: [
          {
            id: 'ed84944f-7d45-4c54-973c-4f5e9696c174',
            to: ['Whalley, Greg', 'Lavorato, John', 'Buy, Richard'],
            sent: '2001-08-23T15:52:44.000Z',
          },
          {
            id: '5d120d9c-3b04-4064-99b8-a56dd343606a',
            to: ['Buy, Richard', 'Glisan, Ben'],
            sent: '2001-10-03T18:35:13.000Z',
          },
          {
            id: '355f82eb-d507-4160-a3dc-55d8907db1dd',
            to: ['Buy, Richard'],
            sent: '2001-02-28T23:39:00.000Z',
          },
          {
            id: 'a2034c12-1aa0-4d71-a909-ad9ca5c7b789',
            to: ['Buy, Richard'],
            sent: '2000-02-11T00:49:00.000Z',
          },
          {
            id: '655b52c3-4d07-44d9-ad91-3363b57dd063',
            to: ['Buy, Richard'],
            sent: '2000-02-11T23:58:00.000Z',
          },
        ],
        asReceiver: [
          {
            id: '973dac60-b462-45bd-a668-bb09deac4386',
            from: 'McMahon, Jeffrey',
            sent: '2001-10-08T22:27:47.000Z',
          },
          {
            id: 'c6bb4562-731b-4b4b-aa47-482e4a9941c7',
            from: 'Buy, Richard',
            sent: '2001-01-31T07:01:00.000Z',
          },
          {
            id: '5559d4ef-e7e5-4634-abac-111d650e0de7',
            from: 'Buy, Richard',
            sent: '2001-10-03T21:38:26.000Z',
          },
          {
            id: '0ca3face-7b64-4b6c-be55-ee6bc12ea187',
            from: 'Glisan, Ben',
            sent: '2001-10-09T19:45:46.000Z',
          },
          {
            id: '58d1c387-d2d8-468b-99e8-0adfcc3d7401',
            from: 'Buy, Richard',
            sent: '2001-01-30T19:01:00.000Z',
          },
          {
            id: '38af6131-b38b-43d5-8f6e-34dcedac3b52',
            from: 'Buy, Richard',
            sent: '2001-01-30T23:58:00.000Z',
          },
          {
            id: 'b9db2813-0fe1-4239-aee9-e1da08ec5922',
            from: 'Buy, Richard',
            sent: '2000-08-04T23:53:00.000Z',
          },
          {
            id: '318256c4-a181-4035-8d97-be4855fb60a9',
            from: 'Buy, Richard',
            sent: '2001-04-23T23:55:00.000Z',
          },
        ],
        name: 'Causey, Richard',
        title: 'Executive Vice President and Chief Accounting Officer',
        color: '#558B2F',
        aliases: [
          'richard causey@enron',
          'richard causey',
          'causey',
          'Causey@enron, Richard',
          'Richard, Causey',
          'causey  richard',
          "'causey,'",
          'richard.causey@enron.com',
          "'causey",
          'rick causey@enron',
          'richard causey/corp/enron@enron',
          'rcausey@enron.com',
        ],
      },
    ],

    // app settings
    densePadding: false,
    darkMode: false,
    themePrimaryColor: '#2196f3',
    themeSecondaryColor: '#f50057',
  },
}
