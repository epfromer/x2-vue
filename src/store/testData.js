export const testWordCloud = [
  {
    tag: 'avici',
    weight: 32,
  },
  {
    tag: 'azurix',
    weight: 523,
  },
  {
    tag: 'backbone',
    weight: 150,
  },
  {
    tag: 'braveheart',
    weight: 29,
  },
]

export const testCustodians = [
  {
    id: 'fastow',
    name: 'Fastow, Andrew',
    title: 'Chief Financial Officer',
    color: '#e91e63',
    senderTotal: 5,
    receiverTotal: 34,
    toCustodians: [{ custodianId: 'lay', total: 1 }],
  },
  {
    id: 'lay',
    name: 'Lay, Kenneth',
    title: 'Founder, CEO and Chairman',
    color: '#ffff00',
    senderTotal: 40,
    receiverTotal: 2690,
    toCustodians: [{ custodianId: 'fastow', total: 1 }],
  },
]

export const testEmailSentByDay = [
  { sent: '1999-01-06', total: 1 },
  { sent: '1999-01-07', total: 2 },
  { sent: '1999-01-08', total: 3 },
]

export const testEmail = [
  {
    id: '692fbb3b-1a4d-4c5b-b8c2-42034586cc56',
    sent: '2001-08-02T02:25:58.000Z',
    sentShort: '2001-08-02',
    from: 'Skilling',
    fromCustodian: 'skilling',
    to: 'allen; Phillip K.; bay; Frank',
    toCustodians: ['lay'],
    cc: 'lay; Kenneth; patrick; Christie',
    bcc: '',
    subject: 'Please Plan to Attend',
    body: 'body 1',
  },
  {
    id: 'f3281cc4-90a9-4dcb-86bd-d705fc847985',
    sent: '2001-08-02T03:21:03.000Z',
    sentShort: '2001-08-02',
    from: 'Skilling',
    fromCustodian: 'skilling',
    to: 'skilling; Jeff; allen; phillip k.',
    toCustodians: ['skilling', 'lay'],
    subject: 'Check out this cool game!',
    cc: 'lay; Kenneth; patrick; Christie',
    bcc: '',
    body: 'body 2',
  },
  {
    id: '5cac6ca4-01e7-4de5-a1d4-806b860e104d',
    sent: '2001-10-12T20:05:56.000Z',
    sentShort: '2001-10-12',
    from: 'Fleming',
    fromCustodian: 'fleming',
    to: 'lay; Kenneth; Amy Taylor (E-mail)',
    toCustodians: ['lay'],
    cc: '',
    bcc: '',
    subject: 'Check out this cool game!',
    body: 'body 3',
  },
]

export const testSearchHistory = [
  {
    id: '862ee9ba-fdd3-4f6c-83d2-c9ffc310277f',
    timestamp: '2020-10-03T13:23:54.955Z',
    entry: '{"skip":0,"limit":50,"sort":"sent","order":1,"to":"ba"}',
  },
  {
    id: 'bdf5571f-ec6a-499b-a0b9-5153a065b04f',
    timestamp: '2020-10-03T13:23:54.516Z',
    entry: '{"skip":0,"limit":50,"sort":"sent","order":1,"to":"ba"}',
  },
]
