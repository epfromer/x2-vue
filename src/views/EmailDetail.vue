<template>
  <div>
    <v-card class="elevation-1">
      <v-card-title>{{ email.subject }}</v-card-title>
      <v-card-text class="text--primary">
        <div>Sent: {{ email.sent }}</div>
        <div>From: {{ email.from }}</div>
        <div>To: {{ email.to }}</div>
        <div>CC: {{ email.cc }}</div>
        <div>BCC: {{ email.bcc }}</div>
        <p></p>
        <span v-html="formattedBody"></span>
      </v-card-text>
      <v-card-actions>
        <v-btn @click="() => this.$router.push('/emaillist')" text>
          Back to list
        </v-btn>
        <v-spacer></v-spacer>
        <!-- <v-btn
          @click="() => setSelected(selected - 1)"
          icon
          :disabled="selected <= 0"
        >
          <v-icon>mdi-arrow-left-bold</v-icon>
        </v-btn>
        <v-btn
          @click="() => setSelected(selected + 1)"
          icon
          :disabled="selected >= savedEmails.length - 1"
        >
          <v-icon>mdi-arrow-right-bold</v-icon>
        </v-btn> -->
      </v-card-actions>
    </v-card>
  </div>
</template>

<script>
import { mapGetters, mapState } from 'vuex'

export default {
  data() {
    return {
      id: this.$route.params.id,
      loading: false,
      email: {},
    }
  },
  mounted() {
    const email = this.getEmailById(this.$route.params.id)
    if (email) {
      this.email = email
    } else {
      this.doFetch()
    }
  },
  methods: {
    // ...mapMutations(['setelected']),
    async doFetch() {
      this.loading = true
      const url = `${process.env.VUE_APP_EMAIL_SERVER}/email/${this.id}`
      console.log(url)
      const resp = await fetch(url)
      resp
        .json()
        .then((resp) => (this.email = resp))
        .catch(() => {}) // TODO: handle errors
        .then(() => (this.loading = false))
      // .then(() => console.log('fetch complete'))
    },
  },
  computed: {
    ...mapGetters(['getEmailById']),
    ...mapState(['emails']),
    formattedBody() {
      return this.email.body ? this.email.body.replace(/\n/g, '<br />') : ''
    },
  },
  watch: {
    // selected(newVal) {
    //   this.email = { ...this.savedEmails[this.selected] }
    // },
  },
}
</script>
