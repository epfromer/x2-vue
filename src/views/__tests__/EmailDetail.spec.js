import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import EmailDetail from '@/views/EmailDetail.vue'
import { emails, state, mutations } from '@/store/mockStore'
import router from '@/router'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('EmailDetail', () => {
  let vuetify
  let store

  beforeEach(() => {
    vuetify = new Vuetify()
    store = new Vuex.Store({
      state,
      mutations
    })
    fetch.resetMocks()
  })

  function doMount(mountType) {
    return mountType(EmailDetail, {
      localVue,
      vuetify,
      store,
      router
    })
  }

  it('should match snapshot', () => {
    const wrapper = doMount(mount)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
