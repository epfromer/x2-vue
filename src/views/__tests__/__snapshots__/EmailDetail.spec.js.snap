// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailDetail should match snapshot 1`] = `
<div>
  <div class="elevation-1 v-card v-sheet theme--light">
    <div class="v-card__title"></div>
    <div class="v-card__text text--primary">
      <div>Sent: </div>
      <div>From: ()</div>
      <div>To: </div>
      <div>CC: </div>
      <p></p> <span></span>
    </div>
    <div class="v-card__actions"><button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default"><span class="v-btn__content">
        Back to list
      </span></button>
      <div class="spacer"></div> <button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-arrow-left-bold theme--light"></i></span></button> <button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-arrow-right-bold theme--light"></i></span></button>
    </div>
  </div>
</div>
`;
